lib/ansible/modules/windows/win_http_proxy.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: win_http_proxy
+version_added: '2.8'
+short_description: Manages proxy settings for WinHTTP
+description:
+- Used to set, remote, or import proxy settings for Windows HTTP Services
+  C(WinHTTP).
+- WinHTTP is a framework used by applications or services, typically .NET
+  applications or non-interactive services, to make web requests.
+options:
+  bypass:
+    description:
+    - A list of hosts that will bypass the set proxy when being accessed.
+    - Use C(<local>) to match hostnames that are not fully qualified domain
+      names. This is useful when needing to connect to intranet sites using
+      just the hostname.
+    - Use C(<-loopback>) to stop automatically bypassing the proxy when
+      connecting through any loopback address like C(127.0.0.1), C(localhost),
+      or the local hostname.
```

Comment:
Did some testing, but was _unable to validate_ this `<-loopback>` part, at least for winhttp.

In particular, on W2016, localhost and 127.0.0.1 destinations appear to ignore the proxy unconditionally, whether `<-loopback>` is in the bypass list or not. I'm also not sure my use of W2016 is a factor, since the MS docs seem to indicate it hasn't materially changed since Vista/2008. I have also not found any support for the idea that `<-loopback>` does anything in the MS docs I reviewed.

Note that I did [find a reference](https://docs.microsoft.com/en-us/windows/desktop/api/wininet/nf-wininet-internetopena) for this construct for wininet, but not for winhttp.

Is this a real thing?

Reply:
It's definitely there for WinINet and I just assumed WinHTTP was the same but it seems like this isn't the case.

As you said I found the reference to `<-loopback>` in the `InternetOpen` function as well as https://blogs.msdn.microsoft.com/ieinternals/2013/10/11/understanding-web-proxy-configuration/. This then has a link to has a link to http://blogs.msdn.com/b/fiddler/archive/2011/02/10/fiddler-is-better-with-internet-explorer-9.aspx but like most MS articles I come across these days it is no longer valid.

Considering this was a feature added in IE9, your testing indicates it doesn't work for WinHTTP, and the lack of docs for this in WinHTTP I will just assume that it isn't valid and remove the references to it in the docs for `win_http_proxy`.

I'll also update the documentation for `win_inet_proxy` to say you need IE9 for this to work.